### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by aaronbuchwald**

### CODE REVIEW

File: plugin/evm/import_tx_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestImportTxVerifyNil(t *testing.T) {
 
 func TestImportTxVerify(t *testing.T) {
 	var importAmount uint64 = 10000000
-	txID := ids.NewID([32]byte{0xff})
+	txID := [32]byte{0xff}
```
Comment (StephenButtolph) [inline plugin/evm/import_tx_test.go:25]:
```
txID := ids.ID{0xff}
```

File: plugin/evm/import_tx_test.go [hunk 2]
```diff
@@ -134,12 +134,12 @@ func TestImportTxSemanticVerify(t *testing.T) {
 
 	importAmount := uint64(1000000)
 	utxoID := avax.UTXOID{
-		TxID: ids.NewID([32]byte{
+		TxID: [32]byte{
```
Comment (StephenButtolph) [inline plugin/evm/import_tx_test.go:137]: `s/[32]byte/ids.ID`

File: plugin/evm/import_tx_test.go [hunk 3]
```diff
@@ -293,12 +293,12 @@ func TestNewImportTx(t *testing.T) {
...
```
Comment (StephenButtolph) [inline plugin/evm/import_tx_test.go:296]: `s/[32]byte/ids.ID`

File: plugin/evm/service.go [hunk 4]
```diff
@@ -19,17 +19,26 @@ import (
 	"github.com/ava-labs/avalanchego/utils/json"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
 )
 
 const (
-	version = "coreth-v0.3.7"
+	version = "coreth"
```
Comment (StephenButtolph) [inline plugin/evm/service.go:27]: Should we have an actual version here?

File: plugin/evm/vm.go [hunk 5]
```diff
@@ -457,12 +448,12 @@ func (vm *VM) ParseBlock(b []byte) (snowman.Block, error) {
 	}
 	blockHash := ethBlock.Hash()
 	// Coinbase must be zero on C-Chain
-	if bytes.Compare(blockHash.Bytes(), vm.genesisHash.Bytes()) != 0 &&
-		bytes.Compare(ethBlock.Coinbase().Bytes(), coreth.BlackholeAddr.Bytes()) != 0 {
+	if !bytes.Equal(blockHash.Bytes(), vm.genesisHash.Bytes()) &&
+		!bytes.Equal(ethBlock.Coinbase().Bytes(), coreth.BlackholeAddr.Bytes()) {
```
Comment (StephenButtolph) [inline plugin/evm/vm.go:452]: Can we just use `!=` here?

File: plugin/evm/vm_test.go [hunk 6]
```diff
@@ -23,18 +23,16 @@ import (
 
 var (
 	testNetworkID    uint32 = 10
-	testCChainID            = ids.NewID([32]byte{'c', 'c', 'h', 'a', 'i', 'n', 't', 'e', 's', 't'})
-	testXChainID            = ids.NewID([32]byte{'t', 'e', 's', 't', 'x'})
-	nonExistentID           = ids.NewID([32]byte{'F'})
+	testCChainID            = ids.ID([32]byte{'c', 'c', 'h', 'a', 'i', 'n', 't', 'e', 's', 't'})
+	testXChainID            = ids.ID([32]byte{'t', 'e', 's', 't', 'x'})
+	nonExistentID           = ids.ID([32]byte{'F'})
```
Comment (StephenButtolph) [inline plugin/evm/vm_test.go:28]: Replace:
```
ids.ID([32]byte{.*})
```
with:
```
ids.ID{.*}
```

File: plugin/evm/vm_test.go [hunk 7]
```diff
@@ -23,18 +23,16 @@ import (
...
 	testTxFee               = uint64(1000)
-	startBalance            = uint64(50000)
 	testKeys         []*crypto.PrivateKeySECP256K1R
 	testEthAddrs     []common.Address // testEthAddrs[i] corresponds to testKeys[i]
 	testShortIDAddrs []ids.ShortID
-	testAvaxAssetID         = ids.NewID([32]byte{1, 2, 3})
-	username                = "Johns"
-	password                = "CjasdjhiPeirbSenfeI13" // #nosec G101
-	ethChainID       uint32 = 43112
+	testAvaxAssetID  = [32]byte{1, 2, 3}
```
Comment (StephenButtolph) [inline plugin/evm/vm_test.go:33]: s/[32]byte/ids.ID

File: plugin/evm/export_tx.go [hunk 8]
```diff
@@ -52,15 +53,15 @@ func (tx *UnsignedExportTx) Verify(
 		return errNilTx
 	case tx.syntacticallyVerified: // already passed syntactic verification
 		return nil
-	case tx.DestinationChain.IsZero():
+	case tx.DestinationChain == ids.Empty:
```
[the remark this answers is not in the document]
Comment (aaronbuchwald) [reply plugin/evm/export_tx.go:56]: Removed

File: plugin/evm/factory.go [hunk 9]
```diff
@@ -3,13 +3,11 @@
...
 
-import (
-	"github.com/ava-labs/avalanchego/ids"
-)
+import "github.com/ava-labs/avalanchego/ids"
 
 // ID this VM should be referenced by
 var (
-	ID = ids.NewID([32]byte{'e', 'v', 'm'})
+	ID = ids.ID([32]byte{'e', 'v', 'm'})
```
[the remark this answers is not in the document]
Comment (aaronbuchwald) [reply plugin/evm/factory.go:10]: changed

File: plugin/evm/import_tx.go [hunk 10]
```diff
@@ -58,15 +58,15 @@ func (tx *UnsignedImportTx) Verify(
 		return errNilTx
 	case tx.syntacticallyVerified: // already passed syntactic verification
 		return nil
-	case tx.SourceChain.IsZero():
+	case tx.SourceChain == ids.Empty:
```
[the remark this answers is not in the document]
Comment (aaronbuchwald) [reply plugin/evm/import_tx.go:61]: removed

File: plugin/evm/import_tx_test.go (hunk 1 above)
Comment (aaronbuchwald) [reply plugin/evm/import_tx_test.go:25]: changed

File: plugin/evm/import_tx_test.go (hunk 2 above)
Comment (aaronbuchwald) [reply plugin/evm/import_tx_test.go:137]: Fixed

File: plugin/evm/import_tx_test.go (hunk 3 above)
Comment (aaronbuchwald) [reply plugin/evm/import_tx_test.go:296]: Fixed

File: plugin/evm/service.go (hunk 4 above)
Comment (aaronbuchwald) [reply plugin/evm/service.go:27]: Changed to v0.3.14

File: plugin/evm/vm.go (hunk 5 above)
Comment (aaronbuchwald) [reply plugin/evm/vm.go:452]: No, byte slices can only be compared to nil. It looks like the comparison between an ID or byte array is valid with `==` or `!=` operators, but byte slices require `bytes.Equal(a,b)`.

File: plugin/evm/vm_test.go (hunk 6 above)
Comment (aaronbuchwald) [reply plugin/evm/vm_test.go:28]: Replaced

File: plugin/evm/vm_test.go (hunk 7 above)
Comment (aaronbuchwald) [reply plugin/evm/vm_test.go:33]: Fixed

File: plugin/evm/vm.go (hunk 5 above)
Comment (StephenButtolph) [reply plugin/evm/vm.go:452]:
```
if blockHash != vm.genesisHash && ethBlock.Coinbase() != coreth.BlackholeAddr {
```

Comment (aaronbuchwald) [reply plugin/evm/vm.go]: Ah ok, just pushed a commit to udpate this.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**